### PR TITLE
wasm-micro-runtime: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **wasm-micro-runtime/***

wasm-micro-runtime is used by harfbuzz/8.x.x to enable experimental wasm features.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/wasm-micro-runtime/all/test_package/test_package.cpp
Comment (RubenRBS) [inline recipes/wasm-micro-runtime/all/test_package/test_package.cpp:1]: We probably don't need a full fledge example in this test package and can simplify it quite a bit :)

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! I have 2 suggestions :)

File: recipes/wasm-micro-runtime/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES CXX)
+
+find_package(wasm-micro-runtime REQUIRED CONFIG)
+
+if(WAMR_NO_INTERP)
+add_executable(${PROJECT_NAME} test_package_no_interp.cpp)
```
[the remark this answers is not in the document]
Comment (toge) [reply recipes/wasm-micro-runtime/all/test_package/CMakeLists.txt:7]: @RubenRBS 
Here is a log:
```
> conan remove -c wasm-micro-runtime; conan create . -o wasm-micro-runtime/\*:with_interp=False -o wasm-micro-runtime/\*:with_aot=True --build=missing --version=1.2.2
Remove summary:
Local Cache
  wasm-micro-runtime/1.2.2#5fe3789b9808b9c93c7c456956ae5452: Removed recipe and all binaries

======== Exporting recipe to the cache ========
WARN: Name containing special chars is discouraged 'wasm-micro-runtime'
wasm-micro-runtime/1.2.2: Exporting package recipe: /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/conanfile.py
wasm-micro-runtime/1.2.2: exports: File 'conandata.yml' found. Exporting it...
wasm-micro-runtime/1.2.2: Calling export_sources()
wasm-micro-runtime/1.2.2: Copied 1 '.yml' file: conandata.yml
wasm-micro-runtime/1.2.2: Copied 1 '.py' file: conanfile.py
wasm-micro-runtime/1.2.2: Copied 1 '.patch' file: 1.2.2-0001-fix-cmake.patch
wasm-micro-runtime/1.2.2: Exported to cache folder: /home/toge/.conan2/p/wasm-2727a3a10e36c/e
wasm-micro-runtime/1.2.2: Exported: wasm-micro-runtime/1.2.2#5fe3789b9808b9c93c7c456956ae5452 (2023-07-17 16:11:20 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux
[options]
wasm-micro-runtime/*:with_aot=True
wasm-micro-runtime/*:with_interp=False

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux


======== Computing dependency graph ========
Graph root
    cli
Requirements
    wasm-micro-runtime/1.2.2#5fe3789b9808b9c93c7c456956ae5452 - Cache

======== Computing necessary packages ========
Requirements
    wasm-micro-runtime/1.2.2#5fe3789b9808b9c93c7c456956ae5452:42d19cbc3f89c50b8e11ebe3fd2f835ff40936b4 - Build

======== Installing packages ========
wasm-micro-runtime/1.2.2: Calling source() in /home/toge/.conan2/p/wasm-2727a3a10e36c/s/src

-------- Installing package wasm-micro-runtime/1.2.2 (1 of 1) --------
wasm-micro-runtime/1.2.2: Building from source
wasm-micro-runtime/1.2.2: Package wasm-micro-runtime/1.2.2:42d19cbc3f89c50b8e11ebe3fd2f835ff40936b4
wasm-micro-runtime/1.2.2: Copying sources to build folder
wasm-micro-runtime/1.2.2: Building your package in /home/toge/.conan2/p/b/wasm-1f1f601bd904f/b
wasm-micro-runtime/1.2.2: Calling generate()
wasm-micro-runtime/1.2.2: Generators folder: /home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/build/Release/generators
wasm-micro-runtime/1.2.2: CMakeToolchain generated: conan_toolchain.cmake
wasm-micro-runtime/1.2.2: CMakeToolchain generated: CMakePresets.json
wasm-micro-runtime/1.2.2: CMakeToolchain generated: ../../../src/CMakeUserPresets.json
wasm-micro-runtime/1.2.2: Generating aggregated env files
wasm-micro-runtime/1.2.2: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
wasm-micro-runtime/1.2.2: Calling build()
wasm-micro-runtime/1.2.2: Apply patch (conan): separate static and shasred build
wasm-micro-runtime/1.2.2: Running CMake.configure()
wasm-micro-runtime/1.2.2: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/toge/.conan2/p/b/wasm-1f1f601bd904f/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/src"
-- Using Conan toolchain: /home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is GNU 13.1.1
-- The CXX compiler identification is GNU 13.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build Configurations:
     Build as target X86_64
     CMAKE_BUILD_TYPE Release
     WAMR Interpreter disabled
     WAMR AOT enabled
     WAMR Fast JIT disabled
     WAMR LLVM ORC JIT disabled
     Libc builtin enabled
     Libc WASI disabled
     Fast interpreter disabled
     Multiple modules disabled
     Bulk memory feature enabled
     SIMD enabled
     Reference types disabled
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/build/Release

wasm-micro-runtime/1.2.2: Running CMake.build()
wasm-micro-runtime/1.2.2: RUN: cmake --build "/home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/build/Release" -- -j12
[  9%] Building C object CMakeFiles/iwasm_static.dir/core/shared/platform/linux/platform_init.c.o
[  9%] Building C object CMakeFiles/iwasm_static.dir/core/shared/platform/common/posix/posix_memmap.c.o
[  9%] Building C object CMakeFiles/iwasm_static.dir/core/shared/platform/common/posix/posix_malloc.c.o
[ 15%] Building C object CMakeFiles/iwasm_static.dir/core/shared/platform/common/posix/posix_thread.c.o
[ 15%] Building C object CMakeFiles/iwasm_static.dir/core/shared/platform/common/posix/posix_socket.c.o
[ 21%] Building C object CMakeFiles/iwasm_static.dir/core/shared/mem-alloc/ems/ems_alloc.c.o
[ 21%] Building C object CMakeFiles/iwasm_static.dir/core/shared/platform/common/posix/posix_time.c.o
[ 25%] Building C object CMakeFiles/iwasm_static.dir/core/shared/mem-alloc/ems/ems_hmu.c.o
[ 28%] Building C object CMakeFiles/iwasm_static.dir/core/shared/mem-alloc/ems/ems_kfc.c.o
[ 31%] Building C object CMakeFiles/iwasm_static.dir/core/shared/mem-alloc/mem_alloc.c.o
[ 34%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/bh_assert.c.o
[ 37%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/bh_common.c.o
[ 40%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/bh_hashmap.c.o
[ 46%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/bh_list.c.o
[ 46%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/bh_log.c.o
[ 50%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/bh_queue.c.o
[ 53%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/bh_vector.c.o
[ 56%] Building C object CMakeFiles/iwasm_static.dir/core/shared/utils/runtime_timer.c.o
[ 59%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/libraries/libc-builtin/libc_builtin_wrapper.c.o
[ 62%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/common/wasm_application.c.o
[ 65%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/common/wasm_c_api.c.o
[ 68%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/common/wasm_exec_env.c.o
[ 71%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/common/wasm_native.c.o
[ 75%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/common/wasm_memory.c.o
[ 78%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/common/wasm_runtime_common.c.o
[ 81%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/common/wasm_shared_memory.c.o
[ 84%] Building ASM object CMakeFiles/iwasm_static.dir/core/iwasm/common/arch/invokeNative_em64_simd.s.o
[ 87%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/aot/aot_intrinsic.c.o
[ 90%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/aot/aot_loader.c.o
[ 93%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/aot/aot_runtime.c.o
[ 96%] Building C object CMakeFiles/iwasm_static.dir/core/iwasm/aot/arch/aot_reloc_x86_64.c.o
[100%] Linking C static library libvmlib.a
[100%] Built target iwasm_static

wasm-micro-runtime/1.2.2: Package '42d19cbc3f89c50b8e11ebe3fd2f835ff40936b4' built
wasm-micro-runtime/1.2.2: Build folder /home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/build/Release
wasm-micro-runtime/1.2.2: Generating the package
wasm-micro-runtime/1.2.2: Temporary package folder /home/toge/.conan2/p/b/wasm-1f1f601bd904f/p
wasm-micro-runtime/1.2.2: Calling package()
wasm-micro-runtime/1.2.2: Running CMake.install()
wasm-micro-runtime/1.2.2: RUN: cmake --install "/home/toge/.conan2/p/b/wasm-1f1f601bd904f/b/build/Release" --prefix "/home/toge/.conan2/p/b/wasm-1f1f601bd904f/p"
-- Install configuration: "Release"
-- Installing: /home/toge/.conan2/p/b/wasm-1f1f601bd904f/p/lib/libvmlib.a
-- Installing: /home/toge/.conan2/p/b/wasm-1f1f601bd904f/p/include/wasm_c_api.h
-- Installing: /home/toge/.conan2/p/b/wasm-1f1f601bd904f/p/include/wasm_export.h
-- Installing: /home/toge/.conan2/p/b/wasm-1f1f601bd904f/p/include/lib_export.h

wasm-micro-runtime/1.2.2: package(): Packaged 1 file: LICENSE
wasm-micro-runtime/1.2.2: package(): Packaged 1 '.a' file: libvmlib.a
wasm-micro-runtime/1.2.2: package(): Packaged 3 '.h' files: wasm_c_api.h, wasm_export.h, lib_export.h
wasm-micro-runtime/1.2.2: Created package revision 0ee8582dfedcbe36f4cfe88dbb50c212
wasm-micro-runtime/1.2.2: Package '42d19cbc3f89c50b8e11ebe3fd2f835ff40936b4' created
wasm-micro-runtime/1.2.2: Full package reference: wasm-micro-runtime/1.2.2#5fe3789b9808b9c93c7c456956ae5452:42d19cbc3f89c50b8e11ebe3fd2f835ff40936b4#0ee8582dfedcbe36f4cfe88dbb50c212
wasm-micro-runtime/1.2.2: Package folder /home/toge/.conan2/p/b/wasm-1f1f601bd904f/p

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    wasm-micro-runtime/1.2.2 (test package): /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/conanfile.py
Requirements
    wasm-micro-runtime/1.2.2#5fe3789b9808b9c93c7c456956ae5452 - Cache

======== Computing necessary packages ========
Requirements
    wasm-micro-runtime/1.2.2#5fe3789b9808b9c93c7c456956ae5452:42d19cbc3f89c50b8e11ebe3fd2f835ff40936b4#0ee8582dfedcbe36f4cfe88dbb50c212 - Cache

======== Installing packages ========
wasm-micro-runtime/1.2.2: Already installed! (1 of 1)

======== Testing the package ========
Removing previously existing 'test_package' build folder: /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release
wasm-micro-runtime/1.2.2 (test package): Test package build: build/gcc-13-x86_64-20-release
wasm-micro-runtime/1.2.2 (test package): Test package build folder: /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release
wasm-micro-runtime/1.2.2 (test package): Writing generators to /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release/generators
wasm-micro-runtime/1.2.2 (test package): Generator 'VirtualRunEnv' calling 'generate()'
wasm-micro-runtime/1.2.2 (test package): Generator 'CMakeDeps' calling 'generate()'
wasm-micro-runtime/1.2.2 (test package): Calling generate()
wasm-micro-runtime/1.2.2 (test package): Generators folder: /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release/generators
wasm-micro-runtime/1.2.2 (test package): CMakeToolchain generated: conan_toolchain.cmake
wasm-micro-runtime/1.2.2 (test package): CMakeToolchain generated: CMakePresets.json
wasm-micro-runtime/1.2.2 (test package): CMakeToolchain generated: ../../../CMakeUserPresets.json
wasm-micro-runtime/1.2.2 (test package): Generating aggregated env files
wasm-micro-runtime/1.2.2 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
wasm-micro-runtime/1.2.2 (test package): Calling build()
wasm-micro-runtime/1.2.2 (test package): Running CMake.configure()
wasm-micro-runtime/1.2.2 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package"
-- Using Conan toolchain: /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- The CXX compiler identification is GNU 13.1.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'wasm-micro-runtime::wasm-micro-runtime'
-- Configuring done (0.1s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release

wasm-micro-runtime/1.2.2 (test package): Running CMake.build()
wasm-micro-runtime/1.2.2 (test package): RUN: cmake --build "/home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/build/gcc-13-x86_64-20-release" -- -j12
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package_no_interp.cpp.o
[100%] Linking CXX executable test_package
/usr/bin/ld: warning: invokeNative_em64_simd.s.o: missing .note.GNU-stack section implies executable stack
/usr/bin/ld: NOTE: This behaviour is deprecated and will be removed in a future version of the linker
[100%] Built target test_package


======== Testing the package: Executing test ========
wasm-micro-runtime/1.2.2 (test package): Running test()
wasm-micro-runtime/1.2.2 (test package): RUN: ./test_package /home/toge/src/conan-center-index/recipes/wasm-micro-runtime/all/test_package/test_package.wasm
Initializing...
Loading binary...
Compiling module...
[03:40:37:203 - 7F4726E7D5C0]: wasm_module_new failed
Creating callback...
Shutting down...
Done.
```

File: recipes/wasm-micro-runtime/all/test_package/test_package.cpp
Comment (toge) [reply recipes/wasm-micro-runtime/all/test_package/test_package.cpp:1]: I agree with you.
I simplified test_package.cpp.

**Review by czoido — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**